backend/services/interfaces/courseService.ts

```diff
@@ -0,0 +1,17 @@
+import {
+  CourseUnitDTO,
+  CreateCourseUnitDTO,
+  UpdateCourseUnitDTO,
+} from "../../types/courseTypes";
+
+interface ICourseService {
+  getCourses(): Promise<Array<CourseUnitDTO>>;
+
+  createCourse(course: CreateCourseUnitDTO): Promise<CourseUnitDTO>;
+
+  updateCourse(id: string, course: UpdateCourseUnitDTO): Promise<CourseUnitDTO>;
+
+  deleteCourse(id: string): Promise<string>;
```

Comment:
could we get some quick comments on what these methods are, like we do in the other service interface files? you're welcome to copy the descriptions in the ticket if that makes it less tedious :)
reason I ask is because there's a couple of interesting things in some of the methods, e.g. delete shifts the indexes of successive units - would be nice to have things like that pointed out